File: src/main/java/com/college/hotlittlepigs/security/CustomPasswordEncoder.java
package com.college.hotlittlepigs.security;

import com.college.hotlittlepigs.user.util.HashUtil;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Component;

@Component
public class CustomPasswordEncoder implements PasswordEncoder {

  @Override
  public String encode(CharSequence rawPassword) {
    return HashUtil.getSecureHash(rawPassword.toString());
  }

  @Override
  public boolean matches(CharSequence rawPassword, String encodedPassword) {
    var hash = HashUtil.getSecureHash(rawPassword.toString());

    return hash.equals(encodedPassword);
  }
}

File: src/main/kotlin/com/college/hotlittlepigs/security/CustomPasswordEncoder.kt
package com.college.hotlittlepigs.security

import com.college.hotlittlepigs.user.util.toSecureHash
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Component

@Component
class CustomPasswordEncoder : PasswordEncoder {
    override fun encode(rawPassword: CharSequence): String = rawPassword.toString().toSecureHash()

    override fun matches(rawPassword: CharSequence, encodedPassword: String): Boolean =
        rawPassword.toString().toSecureHash() == encodedPassword
}